wolfcrypt/src/asn.c

```diff
@@ -16130,8 +16130,8 @@ int ParseCRL(DecodedCRL* dcrl, const byte* buff, word32 sz, void* cm)
     if (ParseCRL_CertList(dcrl, buff, &idx, idx + len) < 0)
         return ASN_PARSE_E;
 
-    if (ParseCRL_Extensions(dcrl, buff, &idx, idx + len) < 0)
-        return ASN_PARSE_E;
+    /* CRL Extensions optional, ignoring errors */
+    ParseCRL_Extensions(dcrl, buff, &idx, idx + len);
```

Comment:
Why are we ignoring errors?

Reply:
If we do not ignore errors then we effectively require CRL extensions. According to [the RFC](https://tools.ietf.org/html/rfc5280#section-5.1) all extensions are optional. Is there another way you would like me to do this?

Reply:
Absolutely it is optional.
So, in ParseCRL_Extensions(), we need to check whether there is an ASN_CONSTRUCTED | ASN_CONTEXT_SPECIFIC | 0 tag and if there isn't then return 0 instead of ASN_PARSE_E.
We can't ignore all the other errors.

Reply:
Can you take a look again if it is alright now?

Reply:
This is right. Thanks!